In [1]:
!pip install -q datascience                                         # Package that is required by pandas profiling
!pip install -q pandas-profiling                                    # Library to generate basic statistics about data

ERROR: Invalid requirement: '#'
ERROR: Invalid requirement: '#'


In [2]:
!pip install -q --upgrade pandas-profiling

In [3]:
#-------------------------------------------------------------------------------------------------------------------------------
import pandas as pd                                                 # Importing for panel data analysis
from pandas_profiling import ProfileReport                          # Importing Pandas Profiling (To generate Univariate Analysis)
pd.set_option('display.max_columns', None)                          # Unfolding hidden features if the cardinality is high
pd.set_option('display.max_colwidth', None)                         # Unfolding the max feature width for better clearity
pd.set_option('display.max_rows', None)                             # Unfolding hidden data points if the cardinality is high
pd.set_option('mode.chained_assignment', None)                      # Removing restriction over chained assignments operations
pd.set_option('display.float_format', lambda x: '%.5f' % x)         # To suppress scientific notation over exponential values
#-------------------------------------------------------------------------------------------------------------------------------
import numpy as np                                                  # Importing package numpys (For Numerical Python)
from datetime import datetime as dt                                 # For datetime funcationality
#-------------------------------------------------------------------------------------------------------------------------------
import plotly.express as ex                                         # Importing Plotly Express for Dynamic Plotting
import plotly.graph_objs as go                                      # Importing Plotly graphs for Dynamic Plotting
import plotly.offline as pyo                                        # Importing offline Express for Dynamic Plotting
from plotly.subplots import make_subplots                           # Importing Plotly Subplots to plot Dynamic subplots
#-------------------------------------------------------------------------------------------------------------------------------
import warnings                                                     # Importing warning to disable runtime warnings
warnings.filterwarnings("ignore")                                   # Warnings will appear only once
#-------------------------------------------------------------------------------------------------------------------------------
from tqdm.notebook import tqdm                                      # To plot progress bar
import plotly.figure_factory as ff                                  # Calling the figure factory to create unique chart types    
#-------------------------------------------------------------------------------------------------------------------------------
import configparser                                                 # Importing Configparser for modifying configration of the script

C:\Users\AK\AppData\Roaming\Python\Python37\site-packages\ipykernel_launcher.py:3: DeprecationWarning: `import pandas_profiling` is going to be deprecated by April 1st. Please use `import ydata_profiling` instead.
  This is separate from the ipykernel package so we can avoid doing imports until


In [6]:

data= pd.read_excel(r'C:\Users\AK\Downloads\Successful projects\earth\earthquake.xlsx')
data.head()



Date      Time  Latitude  Longitude        Type     Depth  \
0  1965-02-01 00:00:00  13:44:18  19.24600  145.61600  Earthquake 131.60000   
1  1965-04-01 00:00:00  11:29:49   1.86300  127.35200  Earthquake  80.00000   
2  1965-05-01 00:00:00  18:05:58 -20.57900 -173.97200  Earthquake  20.00000   
3  1965-08-01 00:00:00  18:49:43 -59.07600  -23.55700  Earthquake  15.00000   
4  1965-09-01 00:00:00  13:32:50  11.93800  126.42700  Earthquake  15.00000   

   Depth Error  Depth Seismic Stations  Magnitude Magnitude Type  \
0          NaN                     NaN    6.00000             MW   
1          NaN                     NaN    5.80000             MW   
2          NaN                     NaN    6.20000             MW   
3          NaN                     NaN    5.80000             MW   
4          NaN                     NaN    5.80000             MW   

   Magnitude Error  Magnitude Seismic Stations  Azimuthal Gap  \
0              NaN                         NaN            NaN   
1              NaN                         NaN            NaN   
2              NaN                         NaN            NaN   
3              NaN                         NaN            NaN   
4              NaN                         NaN            NaN   

   Horizontal Distance  Horizontal Error  Root Mean Square            ID  \
0                  NaN               NaN               NaN  ISCGEM860706   
1                  NaN               NaN               NaN  ISCGEM860737   
2                  NaN               NaN               NaN  ISCGEM860762   
3                  NaN               NaN               NaN  ISCGEM860856   
4                  NaN               NaN               NaN  ISCGEM860890   

   Source Location Source Magnitude Source     Status  
0  ISCGEM          ISCGEM           ISCGEM  Automatic  
1  ISCGEM          ISCGEM           ISCGEM  Automatic  
2  ISCGEM          ISCGEM           ISCGEM  Automatic  
3  ISCGEM          ISCGEM           ISCGEM  Automatic  
4  ISCGEM          ISCGEM           ISCGEM  Automatic

In [35]:
data.describe()

Latitude   Longitude       Depth  Depth Error  \
count 23412.00000 23412.00000 23412.00000   4461.00000   
mean      1.67903    39.63996    70.76791      4.99311   
std      30.11318   125.51196   122.65190      4.87518   
min     -77.08000  -179.99700    -1.10000      0.00000   
25%     -18.65300   -76.34975    14.52250      1.80000   
50%      -3.56850   103.98200    33.00000      3.50000   
75%      26.19075   145.02625    54.00000      6.30000   
max      86.00500   179.99800   700.00000     91.29500   

       Depth Seismic Stations   Magnitude  Magnitude Error  \
count              7097.00000 23412.00000        327.00000   
mean                275.36410     5.88253          0.07182   
std                 162.14163     0.42307          0.05147   
min                   0.00000     5.50000          0.00000   
25%                 146.00000     5.60000          0.04600   
50%                 255.00000     5.70000          0.05900   
75%                 384.00000     6.00000          0.07550   
max                 934.00000     9.10000          0.41000   

       Magnitude Seismic Stations  Azimuthal Gap  Horizontal Distance  \
count                  2564.00000     7299.00000           1604.00000   
mean                     48.94462       44.16353              3.99266   
std                      62.94311       32.14149              5.37726   
min                       0.00000        0.00000              0.00451   
25%                      10.00000       24.10000              0.96875   
50%                      28.00000       36.00000              2.31950   
75%                      66.00000       54.00000              4.72450   
max                     821.00000      360.00000             37.87400   

       Horizontal Error  Root Mean Square  
count        1156.00000       17352.00000  
mean            7.66276           1.02278  
std            10.43040           0.18855  
min             0.08500           0.00000  
25%             5.30000           0.90000  
50%             6.70000           1.00000  
75%             8.10000           1.13000  
max            99.00000           3.44000

**Observations:**
- The average **Depth of an earthquake** recorded is **70.76 KM**.

- Around 25% of records are found to have **Depth of earthquake** less than equal to 14.52 KM.

- Around **50%** of records are found to have **Depth of an earthquake**  less than equal to **33.00 KM**.

- Around **75%** of records are found to have **Depth of an earthquake** less than equal to **33.00 KM**.

- The average records for  **Depth error an earthquake** are found to have **4.99311	KM**.

- Around **25%** records are found to have **Depth error an earthquake**  are recorded to be less than equal to **1.80000	KM**.

- Around **50%** records of **Depth error an earthquake**  are recorded to be less than equal to **3.50000 KM**.

- Around **75%** records of **Depth error an earthquake**  are recorded to be less than equal to **6.30000 KM**.

- The average records **Magnitude of an earthquake** recorded is **5.88253 KM**.

- Around **25%** records of **Magnitude of an earthquake** are recorded to be less than equal to **0.04600 KM**.
|
- The average **Azimuthal Gap** recorded is **5.88253 KM**.

- Around **25%** recrods of **Azimuthal Gap**  are recorded to be less than equal to **24.10000 KM**.

- Around **50%** recrods of **Azimuthal Gap** are recorded to be less than equal to **0.05900 KM**.

- Around **75%** recrods of **Azimuthal Gap** are recorded to be less than equal to **0.07550 KM**.

- Around **50%** recrods of **Magnitude of an earthquake** are recorded to be less than equal to **36.00000	KM**.

- Around **75%** recrods of **Magnitude of an earthquake** are recorded to be less than equal to **54.00000	KM**.


- The average recrods of **Azimuthal Gap** recorded is **5.88253 KM**.

- Around **25%** recrods of **Horizontal Distance** are recorded to be less than equal to **0.96875 KM**.

- Around **50%** recrods of **Horizontal Distance** are recorded to be less than equal to **2.31950	KM**.

- Around **75%** recrods of **Horizontal Distance** are recorded to be less than equal to **4.72450 KM**.

- The average **Root Mean Square** recorded is **5.88253 KM**.

- Around **25%** recrods of **Root Mean Square** are recorded to be less than equal to **0.90000**.

- Around **50%** recrods of **Root Mean Square** are recorded to be less than equal to **1**.

- Around **75%** recrods of **Root Mean Square** are recorded to be less than equal to **1.13**.
 



<a name = Section42></a>
### **Data Information**


In [8]:
data.info(verbose=True, memory_usage='deep', null_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23412 entries, 0 to 23411
Data columns (total 21 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Date                        23412 non-null  object 
 1   Time                        23412 non-null  object 
 2   Latitude                    23412 non-null  float64
 3   Longitude                   23412 non-null  float64
 4   Type                        23412 non-null  object 
 5   Depth                       23412 non-null  float64
 6   Depth Error                 4461 non-null   float64
 7   Depth Seismic Stations      7097 non-null   float64
 8   Magnitude                   23412 non-null  float64
 9   Magnitude Type              23409 non-null  object 
 10  Magnitude Error             327 non-null    float64
 11  Magnitude Seismic Stations  2564 non-null   float64
 12  Azimuthal Gap               7299 non-null   float64
 13  Horizontal Distance         160

**Observations:**
- At high level, we can observe that there are **missing data** in our dataset.
- - **Date** and **Time** are found to be **object** rather they **should be DateTime**.
- The report shows that there in **total 21 features** out of which **9** are **categorical** and **12** is **numerical**.

- Date , Time and ID are found to be highly cardinal in nature. Reference  Pre-Profiling report for further information.

- Time and ID features are found to have uniformity in nature and ID feature also contains unique values. 

- There are no duplicated rows in the dataset.

- As the earthquake depths increase from 0 to 100 KM there is a peak in the distribution which means most of the datapoints are congusted in this area.

- As the earthquake depth increases from 100 KM the frequency reduces.

- Most of the earthquakes range between 5.5KM to 7.0KM magnitude on the Richter scale.

- The most disastrous ones i.e after magnitude 8.0  occur in very lower numbers.


<a name = Section6></a>

---
# **Data Cleaning**
---


In [9]:
# Converting Date to DateTime
data['Date'] = pd.to_datetime(data['Date'], utc=True)
#data['Date'] = data['Date'].dt.strftime('%Y-%m-%d')

# Making Float for data['Depth Seismic Stations'] :- 
data['Depth Seismic Stations'] = pd.to_numeric(data['Depth Seismic Stations'])

# Making Float for data['Depth Seismic Stations'] :- 
data['Root Mean Square'] = pd.to_numeric(data['Depth Seismic Stations'])

In [10]:
# Filling the missing values of Depth Error feature with the mean
data['Depth Error'].fillna(value=data['Depth Error'].mean(), inplace=True)

# Filling the missing values of Depth Error feature with the mean
data['Depth Seismic Stations'].fillna(value=data['Depth Seismic Stations'].mean(), inplace=True)

# Filling the missing values of Depth Error feature with the mean
data['Root Mean Square'].fillna(value=data['Root Mean Square'].mean(), inplace=True)

# Filling the missing values of Depth Error feature with the mean 
data['Magnitude Type'].fillna(value=data['Magnitude Type'].mode()[0], inplace=True)

In [11]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23412 entries, 0 to 23411
Data columns (total 21 columns):
 #   Column                      Non-Null Count  Dtype              
---  ------                      --------------  -----              
 0   Date                        23412 non-null  datetime64[ns, UTC]
 1   Time                        23412 non-null  object             
 2   Latitude                    23412 non-null  float64            
 3   Longitude                   23412 non-null  float64            
 4   Type                        23412 non-null  object             
 5   Depth                       23412 non-null  float64            
 6   Depth Error                 23412 non-null  float64            
 7   Depth Seismic Stations      23412 non-null  float64            
 8   Magnitude                   23412 non-null  float64            
 9   Magnitude Type              23412 non-null  object             
 10  Magnitude Error             327 non-null    float64       

<a name = Section62></a>
### **Dropping the unnecessary features**

In [12]:
# created a list to drop the features stored every feature name in the list itself.
mylist = ['Magnitude Error', 'Magnitude Seismic Stations', 'Horizontal Distance', 'Horizontal Error', 'Azimuthal Gap']

# Dropping the features
for i in tqdm(mylist):
  data.drop(labels=i, axis=1, inplace=True)

  0%|          | 0/5 [00:00<?, ?it/s]

<a name = Section7></a>

---
# **Data Post-Profiling**
---

- We will observe the changes after performing data pre-processing, if present.

In [ ]:
# post_profile=ProfileReport(df=data)
# post_profile.to_file(output_file='Post Profiling Report.html')
# print('Accomplished!')

**Observations:**

- We have successfully cleaned most of the data.

- As we have dropped **5 variables** the total number of features are 16.

- The total number of numerical var. is 7 and categorical var. remians 9. 

- however, as per analysis there is **no significant data loss**.

- The number of missing cells in the dataset has dropped from 29.6% to less than 0.1%. 

- Comparing with the Pre-Profiling Report we can see that there is no huge change in the dataset. 

<a name = Section8></a>

---
# **Exploratory Data Analysis**
---

In [13]:
data['Year'] = data['Date'].dt.year

In [14]:
data['Year']

0        1965
1        1965
2        1965
3        1965
4        1965
5        1965
6        1965
7        1965
8        1965
9        1965
10       1965
11       1965
12       1965
13       1965
14       1965
15       1965
16       1965
17       1965
18       1965
19       1965
20       1965
21       1965
22       1965
23       1965
24       1965
25       1965
26       1965
27       1965
28       1965
29       1965
30       1965
31       1965
32       1965
33       1965
34       1965
35       1965
36       1965
37       1965
38       1965
39       1965
40       1965
41       1965
42       1965
43       1965
44       1965
45       1965
46       1965
47       1965
48       1965
49       1965
50       1965
51       1965
52       1965
53       1965
54       1965
55       1965
56       1965
57       1965
58       1965
59       1965
60       1965
61       1965
62       1965
63       1965
64       1965
65       1965
66       1965
67       1965
68       1965
69       1965
70       1965
71    

In [15]:
data['Year'] = data['Date'].dt.year
data['Day_of_Week'] = data['Date'].dt.day_name()
tmp = data[['Year','Day_of_Week']]

In [16]:
data['Day_of_Week'] = data['Date'].dt.day_name()
data['Day_of_Week']

0           Monday
1         Thursday
2         Saturday
3           Sunday
4        Wednesday
5           Friday
6        Wednesday
7           Friday
8         Saturday
9           Sunday
10          Sunday
11          Sunday
12          Friday
13        Saturday
14         Tuesday
15          Friday
16          Friday
17          Friday
18          Friday
19          Friday
20          Friday
21          Friday
22          Friday
23          Friday
24          Friday
25          Friday
26          Friday
27          Friday
28          Friday
29          Friday
30          Friday
31          Friday
32          Friday
33          Friday
34          Sunday
35          Sunday
36          Sunday
37          Sunday
38          Sunday
39       Wednesday
40       Wednesday
41       Wednesday
42       Wednesday
43       Wednesday
44       Wednesday
45       Wednesday
46          Friday
47          Friday
48          Friday
49          Monday
50        Thursday
51        Thursday
52        Th

In [19]:
tmp = data[['Year','Day_of_Week']]
tmp.head()

Year Day_of_Week
0  1965      Monday
1  1965    Thursday
2  1965    Saturday
3  1965      Sunday
4  1965   Wednesday

In [20]:
tmp=tmp.groupby(by='Year').agg(lambda x:x.value_counts().index[0])
tmp.head()

Day_of_Week
Year            
1965      Monday
1966   Wednesday
1967   Wednesday
1968    Thursday
1969      Monday

In [21]:
tmp = tmp.reset_index()

In [23]:
tmp.head()

Year Day_of_Week
0  1965      Monday
1  1966   Wednesday
2  1967   Wednesday
3  1968    Thursday
4  1969      Monday

In [24]:
data['Day_of_Week'].value_counts()

Saturday     3420
Wednesday    3411
Monday       3405
Thursday     3359
Friday       3296
Sunday       3269
Tuesday      3252
Name: Day_of_Week, dtype: int64

In [25]:
data['Day_of_Week'].value_counts().index[0]

'Saturday'

In [26]:
# Extract labels and values
labels = data['Day_of_Week'].unique()
values = data['Day_of_Week'].value_counts().values

In [27]:
labels

array(['Monday', 'Thursday', 'Saturday', 'Sunday', 'Wednesday', 'Friday',
       'Tuesday'], dtype=object)

In [28]:
values

array([3420, 3411, 3405, 3359, 3296, 3269, 3252], dtype=int64)

In [29]:
# Making the figure
fig = go.Figure(data=[go.Pie(labels=labels, values=values, hole=.8)])
fig

In [30]:
# Making the figure
fig = go.Figure(data=[go.Pie(labels=labels, values=values, hole=.8)])

# Updating the layout
fig.update_layout(height=500, 
                  width=1200, 
                  title_text='Propotion Of Earthquakes On A Certian Day Of Week Over The Years', 
                  title_x=0.5)

# Plotting the figure.
fig

In [32]:
# Preparing the data.
data['Day_of_Week'] = data['Date'].dt.day_name()
tmp = data[['Year','Day_of_Week']]

# Resetting the index.
tmp=tmp.groupby(by='Year').agg(lambda x:x.value_counts().index[0])
tmp = tmp.reset_index()

# Ciming up with the 7 days of a week.
days = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday','Sunday']
days = {k:days[k] for k in range(0,7)}
tmp['Day_of_Week'] = tmp['Day_of_Week'].replace(days)

# Extract labels and values
labels = data['Day_of_Week'].value_counts().index
values = data['Day_of_Week'].value_counts().values

# Making the figure
fig = go.Figure(data=[go.Pie(labels=labels, values=values, hole=.8)])

# Updating the layout
fig.update_layout(height=500, 
                  width=1200, 
                  title_text='Propotion Of Earthquakes On A Certian Day Of Week Over The Years', 
                  title_x=0.5)

# Plotting the figure.
fig

**Observations:**

- The occurrence of earthquakes is in similar proportion across all the days.

In [33]:
data['Day_of_month'] = data['Date'].dt.day
data['Day_of_month']

0         1
1         1
2         1
3         1
4         1
5         1
6         1
7        15
8        16
9        17
10       17
11       24
12       29
13        2
14        2
15        2
16        2
17        2
18        2
19        2
20        2
21        2
22        2
23        2
24        2
25        2
26        2
27        2
28        2
29        2
30        2
31        2
32        2
33        2
34        2
35        2
36        2
37        2
38        2
39        2
40        2
41        2
42        2
43        2
44        2
45        2
46        2
47        2
48        2
49        2
50        2
51        2
52        2
53       15
54       15
55       16
56       17
57       17
58       18
59       18
60       18
61       21
62       23
63       25
64       25
65       26
66       27
67        3
68        3
69        3
70        3
71        3
72        3
73        3
74        3
75        3
76        3
77        3
78        3
79        3
80        3
81        3
82        3
83  

In [34]:
labels = data['Day_of_month'].value_counts().sort_index().index
labels

Int64Index([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16, 17,
            18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31],
           dtype='int64')

In [35]:
values = data['Day_of_month'].value_counts().sort_index().values
values

array([693, 777, 890, 826, 742, 714, 700, 798, 792, 794, 731, 744, 764,
       805, 775, 807, 848, 773, 763, 825, 776, 705, 838, 760, 763, 801,
       711, 800, 679, 672, 346], dtype=int64)

In [36]:
# Preparing data for plotting
data['Day_of_month'] = data['Date'].dt.day
labels = data['Day_of_month'].value_counts().sort_index().index
values = data['Day_of_month'].value_counts().sort_index().values

# Initiating figure for plotting data
fig = go.Figure()

# Adding scatter trace to the figure
fig.add_trace(trace=go.Scatter(x=labels, 
                               y=values,
                               mode='lines+markers'))

# Adding a horizontal line to the graph
fig.add_shape(type="line", 
              x0=labels[0], 
              y0=values.mean(), 
              x1=labels[-1], 
              y1=values.mean(), 
              line=dict(color="Red", 
                        width=2,
                        dash="dashdot"),
              name='Mean')

# Update the layout with some cosmetics
fig.update_layout(height=500, 
                  width=1200, 
                  title_text='Day of the month VS Number Of Earthquakes (per year) Over The Years 1965-2016',
                  xaxis_title = 'Day of the month',
                  yaxis_title = 'Number of earthquakes in a year',
                  title_x=0.5)

# Display the figure
fig

**Observation:**

- On 11th day of the month the total number of recorded earthquakes have been found to be 905 which is maximum.

- We can also see that on 31st day of the month the least number of earthquakes have been recorded and found to be 346.

In [37]:
# Preparing the data.
data['Month'] = data['Date'].dt.month
tmp = data.groupby(by='Month').count()
tmp = tmp.reset_index()[['Month','Magnitude']]

# Making the Figure
fig = go.Figure()

# Adding scatter trace to the figure
fig.add_trace(trace=go.Scatter(x=tmp['Month'], 
                               y=tmp['Magnitude'],
                               mode='lines+markers'))

# Adding a horizontal line to the graph
fig.add_shape(type="line", 
              x0=tmp['Month'][0], 
              y0=tmp['Magnitude'].mean(), 
              x1=tmp['Month'][11], 
              y1=tmp['Magnitude'].mean(), 
              line=dict(color="Red", 
                        width=2,
                        dash="dashdot"),
              name='Mean')

# Updating the layout of the figure.
fig.update_layout(height=500,
                  width=1200,
                  title= 'Number Of Earthquakes Over The months 1965-2016',
                  title_x=0.5,
                  yaxis_title="Number of earthquakes",
                  xaxis_title="Number of Months",
                  xaxis = dict(tickmode = 'linear',
                               tick0 = 0.0,
                               dtick = 1))

# Plotting the figure.
fig.show()

**Observations:**

- We can see aberration in occurrences of earthquakes across different months.

- The maximum number of earthquakes are recorded in March followed by August and December.

- February and June observe the least frequency of earthquakes.

In [38]:
# Preparing data for plotting
data['Year'] = data['Date'].dt.year
labels = data['Year'].value_counts().sort_index().index
values = data['Year'].value_counts().sort_index().values

# Initiating figure for plotting data
fig = go.Figure()

# Adding scatter trace to the figure
fig.add_trace(trace=go.Scatter(x=labels, 
                               y=values,
                               mode='lines+markers'))

# Adding a horizontal line to the graph
fig.add_shape(type="line", 
              x0=labels[0], 
              y0=values.mean(), 
              x1=labels[-1], 
              y1=values.mean(), 
              line=dict(color="Red", 
                        width=2,
                        dash="dashdot"),
              name='Mean')

# Update the layout with some cosmetics
fig.update_layout(height=500, 
                  width=1200, 
                  title_text='Number Of Earthquakes Over The Years 1965-2016',
                  xaxis_title = 'Year',
                  yaxis_title = 'Number of earthquakes in a year',
                  title_x=0.5)

# Display the figure
fig.show()

**Observations:**

- Years 2008-2011 saw the highest number of earthquakes.

- 1966 has the lowest number of recorded earthquakes.

- After 1966, the frequency of earthquakes rises over the years.

In [39]:
# Preparing the data as per the desired format.
tmp = data.groupby(by='Year').mean()
tmp = tmp.reset_index()[['Year','Magnitude']]

# Plotting Figure
fig = go.Figure()

# Adding scatter trace to the figure
fig.add_trace(trace=go.Scatter(x=tmp['Year'], 
                               y=tmp['Magnitude'],
                               mode='lines+markers'))

#Updating the layout of the figure.
fig.update_layout(height=500,
                  width=1200,
                  title= 'Mean Earthquakes Magnitude Over The Years 1965-2016',
                  title_x=0.5,
                  xaxis = dict(tickmode = 'linear',
                               tick0 = 0.0,
                               dtick = 1))

#Plotting the figure.
fig.show()

**Observations:**

- We can see that the average magnitude of earthquakes is **lowest** in **1997**.

- In **1968** the average magnitude of earthquakes is **highest**.

- There is sudden **decline** in the magnitude of Earthquakes after 1972

In [40]:
# Making the data.
tmp = data.groupby(by='Year').std()
tmp = tmp.reset_index()[['Year','Magnitude']]

# Making the Figure.
fig = go.Figure()

# Adding scatter trace to the figure.
fig.add_trace(trace=go.Scatter(x=tmp['Year'], 
                               y=tmp['Magnitude'],
                               mode='lines+markers'))

# Adding a horizontal line to the graph.
fig.add_shape(type="line", 
              x0=1965, 
              y0=tmp['Magnitude'].mean(), 
              x1=2016, 
              y1=tmp['Magnitude'].mean(), 
              line=dict(color="Red", 
                        width=2,
                        dash="dashdot"),
              name='Mean')

# Updating the layout of the figure.
fig.update_layout(height=500,
                  width=1200,
                  title= 'Earthquake Standard Deviation From The Mean Over The Years 1965-2016, Mean SD Shown With Red Line',
                  title_x=0.5,
                  yaxis_title="Standard Deviation From The Mean",
                  xaxis_title="Years from 1965 to 2016",
                  xaxis = dict(tickmode = 'linear',
                               tick0 = 0.0,
                               dtick = 1))

# Plotting the figure.
fig.show()

**Observations:**

- Years 1969, 85, 87, 90, 97 and 2002 have experienced the least deviation from the mean of the total number of earthquakes.

- Years 1965, 68, 71, 73, 77, 82 have experienced the most deviation from the mean of the total number of earthquakes.

In [41]:
# Making the figure.
fig = go.Figure()

# Adding a trace to the figure.
fig.add_trace(go.Scattergeo(lat=data["Latitude"], 
                            lon=data["Longitude"],
                            mode="markers", 
                            showlegend=False, 
                            marker=dict(color="Orange", size=4, opacity=0.5)))

# Updating the geopgraphic parameters.<projection_type="orthographic"> will return a globe.
fig.update_geos(center_lon=-180,
                center_lat=0,
                projection_rotation_lon=-180,
                landcolor="LightGreen",
                oceancolor="LightBlue",
                showocean=True,
                lakecolor="white")

# Updating the base axis.
fig.update_xaxes(tickangle=45)

# Updating the layout.
fig.update_layout(height=500, 
                  width=1200,
                  template="plotly_white",
                  margin=dict(r=20, t=25, b=40, l=20),
                  title= 'All earthquakes that happened from 1965-2016',
                  title_x=0.5,
                  title_y=1)

# Plotting the figure.
fig.show()

**Observations:**

- Oceanic Region experiences a lot more seismic activity than mainland regions.

- This corresponds to the tectonic plate movements below the ocean bed.

In [42]:
# Making the configration object of configparser class.
config = configparser.ConfigParser()
config = config.read('config.ini')

# Adding a mapbox token.
mapbox_token = 'pk.eyJ1IjoiY2xhdWNvdCIsImEiOiJja2h4MTIxd2UwMzNzMnlvNzVycXN2dW14In0.J_KSzOY3YGDzozRdHvMJgQ'

# Creating the figure.
fig = go.Figure(go.Scattermapbox(lat=data['Latitude'], 
                                 lon=data['Longitude'],
                                 mode='markers',
                                 marker = go.scattermapbox.Marker(size=data['Magnitude'],
                                                                  colorscale='IceFire',
                                                                  color=data['Magnitude'],
                                                                  showscale=True,
                                                                  colorbar={'title': 'Magnitude', 
                                                                            'titleside': 'top', 
                                                                            'thickness': 4, 
                                                                            'ticksuffix': ''}),
                                 customdata=np.stack((pd.Series(data['Date'].dt.year),
                                                      data['Magnitude'],
                                                      data['Depth']),
                                                     axis=-1),
                                 hovertemplate=
                                 """
                                 <extra></extra>
                                 <em>%{customdata[0]}  </em><br>
                                 Magnitude  %{customdata[1]}<br>
                                 Depth  %{customdata[2]}<br>
                                 """))

# updating the layout
fig.update_layout(height=500,
                  width=1200, 
                  title_text='Areas where earthquakes occur most frequently', 
                  title_x=0.5,
                  mapbox=dict(accesstoken=mapbox_token,
                              center=go.layout.mapbox.Center(lat=25,
                                                             lon=142), 
                              zoom=1))

**Observations:**

- We can see that most of the earthquakes occur on the sea area whereas the lands experience less amount of earthquakes.

- Most of the earthquakes have been recorded at the centre of the earth i.e, the areas connecting to the North Pacific Ocean with an average magnitude of 7 KM.


In [43]:
# Creating a Figure.
#fig = go.Figure()

fig = make_subplots(rows=1, 
                    cols=3, 
                    subplot_titles=('Magnitude vs Magnitude Type', 
                                    'Magnitude vs Status', 
                                    'Root Mean Square vs Status'))

# Adding first trace of bar plot
fig.add_trace(trace=go.Bar(x=data['Magnitude'], 
                           y=data['Magnitude Type'], 
                           orientation='h', 
                           marker=dict(color='rgba(238, 227, 231, 1.0)',
                                       line=dict(color='rgba(148, 42, 43, 1.0)', 
                                                 width=3))), 
              row=1, col=1)
              
# Adding Traces
fig.add_trace(trace=go.Bar(x=data['Magnitude'], 
                           y=data['Status'], 
                           orientation='h', 
                           marker=dict(color='rgba(238, 227, 231, 1.0)',
                                       line=dict(color='rgba(148, 42, 43, 1.0)', 
                                                 width=3))), 
              row=1, col=2)

# Adding Traces
fig.add_trace(trace=go.Bar(x=data['Root Mean Square'], 
                           y=data['Status'], 
                           orientation='h', 
                           marker=dict(color='rgba(238, 227, 231, 1.0)',
                                       line=dict(color='rgba(148, 42, 43, 1.0)', 
                                                 width=3))), 
              row=1, col=3)


# Updating the layout.
fig.update_layout(height=500, 
                  width=1200, 
                  title_text='Magnitude vs Magnitude Type and Status vs Magnitude and Root Mean Square', 
                  title_x=0.5,  
                  showlegend=False)

# Adding subplot layout axis labels
fig['layout']['xaxis']['title']='Magnitude'
fig['layout']['yaxis2']['title']='Status'
fig['layout']['xaxis2']['title']='Magnitude'
fig['layout']['yaxis']['title']='Magnitude Type'
fig['layout']['xaxis3']['title']='Root Mean Square'
fig['layout']['yaxis2']['title']='Status'

# Plotting the curve.
fig.show()

**Observations:**

- MW is recorded to have the highest amount of magniitude and that is 6.5 KM.

- Reviewed status has more root mean square value compared to automatic status.

- It has been recorded that Reviewed status has more Magnitude value compared to automatic status.


<a name = Section9></a>

---
# **Summarization**
---

- **<h4>Conclusion</h4>**

    - From this case study it was analyzed that **Latitude** **Longitude**, **Depth seismic stations** and **Magnitude** are the most important features whereas, **Magnitude Error**, **Magnitude Seismic Stations**, **Horizontal Distance**, **Horizontal Error**, **Azimuthal Gap** were not that much important.
 
    - The 11th day of the month has experienced the maximum number of earthquakes whereas the 31st day has experienced the least number of earthquakes.
 
    - On the 3rd and the 8th month, it has been noticed that the number of earthquakes reported was maximum.
 
    - It was analysed that the Magnitude Type **MW** has the highest amount of magnitude whereas **MD** had the least magnitude value.


-  **<h4>Actionable Insights</h4>**

    - Manually reviewed status has performed better over automatic checks hence, manual evaluation of earthquake magnitude is advisable.   
 
    - As the oceanic region experiences a lot more seismic activity than mainland regions and most of the earthquakes have been reported to the lands connected to the **north pacific ocean**, it is advised not to perform any project or execution-based task in the 3rd and 8th month of the year in order to avoid physical consequences.